src/node.rs

```diff
@@ -483,24 +483,27 @@ impl<P: KeyTrait + Clone, N: Version> NodeTrait<N> for Node48<P, N> {
     }
 
     fn delete_child(&self, key: u8) -> Self {
-        let pos = self.keys.get(key as usize).unwrap();
+        let pos = self.keys[key as usize];
+        assert!(pos != u8::MAX);
         let mut new_node = self.clone();
-        new_node.keys.erase(key as usize);
-        new_node.children.erase(*pos as usize);
-        new_node.num_children -= 1;
+        new_node.keys[key as usize] = u8::MAX;
+        new_node.children[pos as usize] = None;
+        new_node.child_bitmap &= !(1 << pos);
 
         new_node.update_version_to_max_child_version();
         new_node
     }
 
     fn find_child(&self, key: u8) -> Option<&Arc<N>> {
-        let idx = self.keys.get(key as usize)?;
-        let child = self.children.get(*idx as usize)?;
-        Some(child)
+        let idx = self.keys[key as usize];
+        if idx == u8::MAX {
+            return None;
+        }
+        Some(self.children[idx as usize].as_ref().unwrap())
     }
 
     fn num_children(&self) -> usize {
-        self.num_children as usize
+        self.child_bitmap.count_ones() as usize
```

Comment:
Since this field is now a bitmap the amount of children is the amount of ones in the integer.